### ISSUE TITLE
vmap with rbg prng causes unexpected collectives

### ISSUE THREAD

**Issue by eamartin** [jax-ml/jax#34497]
### Description

It is known and documented that vmap when using the RBG PRNG uses the first prng key for the full-prng generation:
https://docs.jax.dev/en/latest/jax.random.html#advanced-rng-configuration
https://github.com/jax-ml/jax/pull/20094

I assume this is to increase the performance of XLA RBG by calling it with more generated bits per key.

I was surprised to discover that "reuse the first prng key" can cause collectives when the prng key is sharded, as the first key must be broadcast to all hosts. For multi-slice TPU jobs, this can be an expensive Megascale collective. This is doubly surprising if you don't know about the RBG behavior - one may think "why in the world is my properly sharded prng call with vmap doing Megascale collectives".

I believe Jax could still maximize XLA RBG's generated bits per key while avoiding this collective if instead of using the first entry of the global prng matrix instead using the first entry of each shard.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.8.2.dev20251203
jaxlib: 0.8.2.dev20251203
numpy:  2.3.3
python: 3.12.3 (main, Jan  8 2026, 11:30:50) [GCC 13.3.0]
```

**Comment by froystig** [comment]
@yashk2810 and I drafted the change linked above, which ought to generate from a per-shard key, *if* the batch axis of keys is explicitly sharded. Are you sharding explicitly, or able to do so prior to drawing the random sample?

The change still needs review and a decision, since this is a bits-altering change. It also violates sharding invariance, departing from the table in the docs that you linked, but only in the explicit sharding case.
